CONTRIBUTING.md

```diff
@@ -0,0 +1,29 @@
+# Contributing
+
+## Creating a new sub-project
+
+The project is a monorepo that uses NX to manage it's components.
+
+The project has some NX generators for creating [hooks](https://docs.openfeature.dev/docs/reference/concepts/hooks) and [providers](https://docs.openfeature.dev/docs/reference/concepts/provider).
+
+`npm run generate-hook` <- generates a hook component
+`npm run generate-provider` <- generates a provider component
+
+The script will create the basic code scaffolding, and infrastructure to publish the artifact.
+
+## Documentation
+
+Any published components must have documentation in their root directory, explaining the basic purpose of the component as well as installation and usage instructions.
+Instructions for how to develop a component should also be included (required system dependencies, instructions for testing locally, etc).
+
+## Testing
+
+Any published components must have reasonable testing.
+The NX scaffolding will generate stub tests for you when you create your project.
```

Comment:
Maybe show the tests commends:

 - npm run test
 - npx nx test {{ COMPONENT NAME }}